lib/core/src/lib/datatable/components/datatable/datatable.component.ts

```diff
@@ -639,8 +639,8 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
             if (current && column.key === current.key) {
                 newDirection = current.direction?.toLowerCase() === 'asc' ? 'desc' : 'asc';
             }
-            this.sorting = [column.key, newDirection];
-            this.data.setSorting(new DataSorting(column.key, newDirection));
+            this.sorting = [column.key, newDirection, { numeric: true }];
+            this.data.setSorting(new DataSorting(column.key, newDirection, { numeric: true }));
```

Comment:
the problem is that there is no way to check it, because column type for integers is as "text"

Reply:
Additionally if I remember correctly, there was a point during our call with Agnieszka that maybe it could be good idea to have sorted mixed values like: 

a1
a10
file1
file2
file3

So you can see in above example that it sorts first by strings ('a' before 'file') and then it additionally sorts it by number (like 1, 10 or 1, 2, 3). 

In that case "numeric" field allows for this kind of sorting.

Reply:
Okay in that case it's fine for me, maybe it would be also worth to ask someone from HX to review to ensure we're not breaking their side

Reply:
the numbers should be used only when the column type is set to 'number'